### ISSUE TITLE
dataset API study endpoint does not return all studies

### ISSUE THREAD

**Issue by aclum** [microbiomedata/nmdc-server#1592]
The dataset studies endpoint with no arguments is missing some studies. From spot checking the missing studies appear to be those that are part_of another study. This was reported by KBase, I've redirected them to use the runtime API but we should discuss what to do here.

base_url = “https://data.microbiomedata.org/api”
endpoint = “study/search”

studies that are missing from the API response body
{'nmdc:sty-11-33fbta56',
 'nmdc:sty-11-34xj1150',
 'nmdc:sty-11-5tgfr349',
 'nmdc:sty-11-aygzgv51',
 'nmdc:sty-11-dcqce727',
 'nmdc:sty-11-e4yb9z58',
 'nmdc:sty-11-hht5sb92',
 'nmdc:sty-11-kfvmk798',
 'nmdc:sty-11-n7mtj961',
 'nmdc:sty-11-pzmd0x14',
 'nmdc:sty-11-vh2hty57',
 'nmdc:sty-11-wbc14h22'}

**Comment by eecavanna** [comment]
Here is a sequence of `curl` commands people can use to reproduce the symptom.


1. Get one biosample whose `study_id` is `nmdc:sty-11-33fbta56`. The response consists of a biosample.
   ```shell
   curl -X 'POST' \
     'https://data.microbiomedata.org/api/biosample/search?offset=0&limit=1' \
     -H 'accept: application/json' \
     -H 'Content-Type: application/json' \
     -d '{
     "conditions": [{"table": "biosample", "field": "study_id", "op": "==", "value": "nmdc:sty-11-33fbta56"}]
   }'
   ```
2. Get "all" studies. The response consists of 29 studies, none of which has an `id` of `nmdc:sty-11-33fbta56`.
   ```shell
   curl -X 'POST' \
     'https://data.microbiomedata.org/api/study/search?offset=0&limit=50' \
     -H 'accept: application/json' \
     -H 'Content-Type: application/json' \
     -d '{
     "conditions": []
   }'
   ```
3. Get studies having an `id` of `nmdc:sty-11-33fbta56`. The response consists of a study having that `id`.
   ```shell
   curl -X 'POST' \
     'https://data.microbiomedata.org/api/study/search?offset=0&limit=50' \
     -H 'accept: application/json' \
     -H 'Content-Type: application/json' \
     -d '{
     "conditions": [{"table": "study", "field": "id", "op": "==", "value": "nmdc:sty-11-33fbta56"}]
   }'
   ```

In the description of `curl` command number 2, I put the word "all" in quotes because, based on the response to `curl` command number 3, I do not think `curl` command number 2 really returns all studies.

**Comment by naglepuff** [comment]
Good catch. I am able to find the study on the data portal (dev). The requests made from the front end add a filter (based on type, e.g. research_study or consortium). Not sure how that affects the logic, but we should definitely look into this.

**Comment by naglepuff** [comment]
Hmm.... @eecavanna @aclum the endpoint runs some logic to nest studies using the "part_of" relationship. The "29" studies that you see in the JSON refer to 29 parent studies. There's another top-level property of the response object called "total," which refers to the total number of studies in the response.

Using the swagger page at https://data.microbiomedata.org/api/docs, I tested the studies/search endpoint using a limit of 50, and I was able to find the first few IDs listed in the initial issue. In particular, the one @eecavanna also couldn't find was nested until the study with the ID `nmdc:sty-11-cytnjc39`. The relationship is through a property on the parent study called `children`. The value for "total" in this case was 44. This nesting is used by the data portal frontend to properly group studies by the part_of relationship.

**Comment by aclum** [comment]
I'm going to close this as won't fix and we'll need to reconcile how to handle this with a unified backend later.
